.github/workflows/main-branch-build.yml

```diff
@@ -0,0 +1,32 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs
+
+
+name: Event Pro CI
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build:
+    name: Main Branch Build
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:

```

Comment:
For this project, we really dont need to use the matrix strategy, since we are only going to build this app once, and then deploy that to vercel. If we care about a specific node version, we specify just that version. Having our app built for multiple node versions would be more important if someone was going to download and run our app locally.